app/actions/remote/channel.ts

```diff
@@ -113,14 +118,27 @@ export const fetchMyChannelsForTeam = async (serverUrl: string, teamId: string,
             return result;
         }, []);
 
-        if (!fetchOnly) {
+        // if (!fetchOnly) {
+        if (true) {
             const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
             const modelPromises: Array<Promise<Model[]>> = [];
             if (operator) {
-                const prepare = await prepareMyChannelsForTeam(operator, teamId, channels, memberships);
-                if (prepare) {
-                    modelPromises.push(...prepare);
+                const prepareChannelModels = await prepareMyChannelsForTeam(operator, teamId, channels, memberships);
```

Comment:
_Just an opinion_: 
Here, we are hijacking the function `fetchMyChannelsForTeam` to retrieve the categories, however, it only returns the `channels and memberships`.  Instead, we can create a separate `fetchMyChannelCategories`  function and just call without awaiting it at around line 100 in the spirit of maintaining a single purpose for `fetchMyChannelsForTeam` function.

Reply:
I disagree with this comment, as same would be true for channel members, this is fine as it relates to my channels

Reply:
In any case I would await, batch the results and also add it to what gets returned by this function. Or extract it completely and call it from outside this function.

Now if we ever going to fetch the categories qhen we fetch myChannels & members I see no reason to decouple it